### ISSUE TITLE
Raising of 409 Conflict code

### ISSUE THREAD

**Issue by vdusek** [encode/django-rest-framework#6124]
## Description
When a unique constraint is violated status code is ```400 Bad Request```. Instead, I'd like ```409 Conflict``` to be raised.

## Expected behavior
I am able to subclass some of the DRF exception to raise my own ```status_code``` and my own error ```detail```.

## Actual behavior
When I create a subclass of [ValidationError](https://github.com/encode/django-rest-framework/blob/master/rest_framework/exceptions.py#L142) I can configure my own ```detail``` and it works fine. However when I want to configure my own ```status_code```, it doesn't work at all and ```400 Bad Request``` is raised anyway.

When I create a subclass of [APIException](https://github.com/encode/django-rest-framework/blob/master/rest_framework/exceptions.py#L98) my ```status_code``` is raised properly, but ```detail``` is wrong formated. It's not in a ```{'key': ['error detail', ...]}``` format.

**Comment by vdusek** [comment]
@dparalen cc

**Comment by tomchristie** [comment]
Sure. Worth having a search through the issues for ‘409’. Short version is that we can’t *really* provide different error codes for different types of field error, because serializers can include multiple field errors.

**Comment by dparalen** [comment]
True. We had a light of hope by trying to address this as a special, "quick" failure (as can be seen in the PR), sacrifying the other possible form field errors in favor of a conflict being reported.

**Comment by omkar-dsd** [comment]
Please mention it in documentation of DRF.

**Comment by xordoquy** [comment]
It is already said [in the documentation](http://www.django-rest-framework.org/api-guide/serializers/#raising-an-exception-on-invalid-data) that validation errors returns a 400 error code.

**Comment by omkar-dsd** [comment]
It is not clear from the documentation that the default code (400) cannot be overridden.
